spec/money_spec.rb

```diff
@@ -814,35 +800,139 @@ def m.amount
       it "maintains the currency" do
         expect(rounded.currency).to eq Money::Currency.new('NZD')
       end
+    end
+
+    context "when using a subclass of Money" do
+      let(:special_money_class) { Class.new(Money) }
+      let(:money) { special_money_class.new(15.75, 'NZD') }
+
+      it "preserves the class in the result" do
+        expect(rounded).to be_a special_money_class
+      end
+    end
+  end
+
+  describe "#round" do
+    it_behaves_like "rounding" do
```

Comment:
this is a weird construct — what's the purpose of defining new examples inside a shared example? normally you'd only use the block to redefine some of the `let` params

Reply:
That is because those new examples are using the same `money` `let`. :grin: 
It can be avoided with duplication. Will do.